packages/shopify-api/lib/utils/shop-admin-url-helper.ts

```diff
@@ -0,0 +1,74 @@
+// This class assumes the legacy url has protocol stripped already
+// Converts admin.shopify.com/store/my-shop to my-shop.myshopify.com
+export function shopAdminUrlToLegacyUrl() {
+  return (shopAdminUrl: string): string | null => {
+    const isShopAdminUrl = shopAdminUrl.split('.')[0] === 'admin';
+
+    if (!isShopAdminUrl) {
+      return null;
+    }
+
+    const regex = new RegExp(`admin\\..+/store/(.+)`);
+    const matches = shopAdminUrl.match(regex);
+
+    if (matches && matches.length === 2) {
+      const shopName = matches[1];
+      const isSpinUrl = shopAdminUrl.includes('spin.dev/store/');
+
+      if (isSpinUrl) {
+        return spinAdminUrlToLegacyUrl(shopAdminUrl);
+      } else {
+        return `${shopName}.myshopify.com`;
+      }
+    } else {
+      return null;
+    }
+  };
+}
+
+// Converts my-shop.myshopify.com to admin.shopify.com/store/my-shop
+export function legacyUrlToShopAdminUrl() {
+  return (legacyAdminUrl: string): string | null => {
```

Comment:
I wasn't quite sure about this since I modelled it after sanitizeHost()

https://github.com/Shopify/shopify-api-js/blob/bd3198d592bc1ef9e61e5e0d581ce8397702ff2f/packages/shopify-api/lib/utils/shop-validator.ts#L42-L43

I'm not familiar with TS, so I thought we needed this so it can be used as a part of `shopify.utils.method`?